src/ctapipe/core/traits.py

```diff
@@ -72,8 +74,48 @@
 flag = traitlets.config.boolean_flag
 
 
+class AstroQuantity(TraitType):
+    """A trait containing an ``astropy.units`` quantity."""
+
+    def __init__(self, physical_type=None, **kwargs):
+        super().__init__(**kwargs)
+        self.physical_type = physical_type
```

Comment:
This should probably do something more nuanced than simply trusting that user input is of type `astropy.unit`...

like 
```
if physical_type and isinstance(physical_type,u.unit):
     self.physical_type = physical_type
elif not physical_type:
     self.physical_type = physical_type
else:
    raise ValueError("if given physical_type must be of type astropy.unit, was {type(physical_type)}")
```

Reply:
The expected type is u.PhysiclType not u.Unit.

However, I guess it would be nice to allow unit input and call unit.physical_type in that case

Reply:
Good point. I added support for both `units.PhysicalType` and `units.Unit`.